raft/storage.go

```diff
@@ -57,6 +57,17 @@ type Storage interface {
 	Snapshot() (pb.Snapshot, error)
 }
 
+// InitialState is the return value of the Storage.InitialState method.
+type InitialState struct {
+	HardState pb.HardState
+	ConfState pb.ConfState
+
+	// AppliedIndex is the last index which has been applied to the state machine,
+	// for implementations which persist the state machine to disk. If the state
+	// machine is not persistent, this field may be zero.
+	AppliedIndex uint64
```

Comment:
hmm... i am not sure if we should put this into storage. or we should make this a argument when starts raft.

usually the applier and the storage are separated. so the normal restart process would be:
1. recover the applier and the the applied index
2. start raft with raft.storage and applied index.